src/storage/mongodb/AssetStorage.ts

```diff
@@ -15,10 +15,14 @@ const MODULE_NAME = 'AssetStorage';
 
 export default class AssetStorage {
   public static async getAsset(tenant: Tenant, id: string = Constants.UNKNOWN_OBJECT_ID,
-      params: { withSiteArea?: boolean } = {}, projectFields?: string[]): Promise<Asset> {
+    params: { withSiteArea?: boolean, siteIDs?: string[], issuer?: boolean, dynamicOnly?: boolean, usesPushAPI?:boolean } = {}, projectFields?: string[]): Promise<Asset> {
     const assetsMDB = await AssetStorage.getAssets(tenant, {
       assetIDs: [id],
-      withSiteArea: params.withSiteArea
+      withSiteArea: params.withSiteArea,
+      siteIDs: params.siteIDs,
+      issuer: params.issuer,
+      dynamicOnly: params.dynamicOnly,
+      usesPushAPI: params.usesPushAPI
```

Comment:
I dont aWeird these 2 filters: dynamicOnly and usesPushAPI
I don't see a user's role which you want it to have only dynamic asset or push api usage.